mautic/mautic
# Improve boolean import/API support

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
When creating leads by imports or APIs, boolean values that don't use "true" or "false" (etc) end up being stored as FALSE, instead of NULL (a more accurate value for a non-answer). This can cause a lot of confusion. This fixes that and also supports values like "T" and "F" which are obviously representative.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a boolean field.
2. Import a CSV with one lead value of "T".
3. Open the lead imported and you'll see "No" (false value) selected.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a boolean field.
3. Import a CSV with one lead value of "T".
4. Open the lead imported and you'll see "Yes" (true value) selected.
